plugin/pkg/admission/priority/admission.go

```diff
@@ -203,6 +207,9 @@ func (p *PriorityPlugin) validatePriorityClass(a admission.Attributes) error {
 	if pc.Value > HighestUserDefinablePriority {
 		return admission.NewForbidden(a, fmt.Errorf("maximum allowed value of a user defined priority is %v", HighestUserDefinablePriority))
 	}
+	if strings.HasPrefix(pc.Name, SystemPriorityClassPrefix) {
```

Comment:
If a priority class is created while this plugin is not enabled, does this prevent update/deleting it?

Reply:
+1, validate in api server?

Reply:
> If a priority class starting with "system-" is created while this plugin is not enabled, does this prevent update/deleting it?

It does not prevent deletion, but it does prevent updates. I think this should be fine. We don't allow updating names of priority classes anyway.
I move the new logic to the API side.